generator_test.go

```diff
@@ -142,10 +156,53 @@ func testNewV1MissingNetworkFaultyRand(t *testing.T) {
 	}
 }
 
+func testNewV1Ordering(t *testing.T) {
+	u1 := Must(NewV1())
+	u2 := Must(NewV1())
+
+	n1, err := u1.UnixNanoOf()
+	if err != nil {
+		t.Fatal(err)
+	}
+	n2, err := u2.UnixNanoOf()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if n1 > n2 {
+		t.Errorf("generated older UUID: %s > %s", u1, u2)
+	}
+
+	t1, err := u1.TimeOf()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t2, err := u2.TimeOf()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if t1.After(t2) {
+		t.Errorf("time comparison failed: %s (%s) > %s (%s)", u1, t1, u2, t2)
+	}
+}
+
 func testNewV2(t *testing.T) {
 	t.Run("Basic", testNewV2Basic)
 	t.Run("DifferentAcrossCalls", testNewV2DifferentAcrossCalls)
 	t.Run("FaultyRand", testNewV2FaultyRand)
+	t.Run("TestTimeOfErrors", testTimeOfErrors)
+}
+
+func testTimeOfErrors(t *testing.T) {
```

Comment:
This test does not test a property of `NewV2`, so I don't think it should be a subtest of `testNewV2`.